console/src/components/UIKit/theme/index.ts

```diff
@@ -1,6 +1,29 @@
 // Theme specification for the Design-System.
-
-const colors = {
+type ColorName =
+  | 'red'
+  | 'green'
+  | 'blue'
+  | 'orange'
+  | 'yellow'
+  | 'grey'
+  | 'black'
+  | 'white'
+  | 'transparent'
+  | 'tab';
+
+type ColorType =
+  | 'original'
+  | 'primary'
+  | 'secondary'
+  | 'hover'
+  | 'light'
+  | 'border'
+  | 'text'
+  | 'tab';
+
+type Color = Partial<Record<ColorType, string>> | any;
```

Comment:
I couldn't find a proper way to handle this with `Union Types` or `Partial`.  Because there are color objects as both `record` and `string`.  What do you think ?

Reply:
Generally, I'd skip proving typings manually for `colors`, `icon`, `alertbox`, `button` -- TypeScript can infer those types on its own, and by typing it manually, we may accidentally break something. 

However, for parts of the theme, that have aliases (eg `radii`) things are more complicated, and we do need typings. 

So for those first four (`colors`, `icon`, `alertbox`, `button`) I'd remove typings; the rest looks okay. 

One suggestion: 
Could you change `FontWeight` -> `FontWeightKey`,  `Element` -> `FontSizeKey`, `Space` -> `SpaceKey`, etc. ?
That way, they won't be confused with the types of actual objects.

Reply:
I removed types from the first four objects. But we may need some of them as typed once we start to convert other components. I will let you know if something like that occurs. 

Apart from that, I think this is ready.

Reply:
If we need them, for example, the type of `colors` object, we could do this:

<img width="536" alt="Screenshot 2020-04-09 at 13 54 47" src="https://user-images.githubusercontent.com/9019397/78892467-b086cb00-7a69-11ea-95b3-f029abf1038f.png">